src/borg/crypto/key.py

```diff
@@ -432,13 +432,33 @@ def decrypt_key_file(self, data, passphrase):
         unpacker = get_limited_unpacker('key')
         unpacker.feed(data)
         data = unpacker.unpack()
-        enc_key = EncryptedKey(internal_dict=data)
-        assert enc_key.version == 1
-        assert enc_key.algorithm == 'sha256'
-        key = passphrase.kdf(enc_key.salt, enc_key.iterations, 32)
-        data = AES(key, b'\0'*16).decrypt(enc_key.data)
-        if hmac_sha256(key, data) == enc_key.hash:
+        encrypted_key = EncryptedKey(internal_dict=data)
+        if encrypted_key.version == 1:
+            return self.decrypt_key_file_v1(encrypted_key, passphrase)
+        elif encrypted_key.version == 2:
+            return self.decrypt_key_file_v2(encrypted_key, passphrase)
+        else:
+            raise Exception('Your borg key is stored in an unsupported format. Try using a newer version of borg')
+
+    def decrypt_key_file_v1(self, encrypted_key, passphrase):
+        assert encrypted_key.algorithm == 'sha256'
+        key = passphrase.kdf(encrypted_key.salt, encrypted_key.iterations, 32)
+        data = AES(key, b'\0'*16).decrypt(encrypted_key.data)
+        if hmac_sha256(key, data) == encrypted_key.hash:
             return data
+        return None
+
+    def decrypt_key_file_v2(self, encrypted_key, passphrase):
+        enc_key, mac_key = passphrase.argon2(
```

Comment:
maybe also add a algorithm check above this like in v1 in case we ever want to change?